planner/core/rule_aggregation_push_down.go

```diff
@@ -305,9 +305,10 @@ func (a *aggregationPushDownSolver) pushAggCrossUnion(agg *LogicalAggregation, u
 	// this will cause error during executor phase.
 	for _, key := range unionChild.Schema().Keys {
 		if tmpSchema.ColumnsIndices(key) != nil {
-			proj := a.convertAggToProj(newAgg)
-			proj.SetChildren(unionChild)
-			return proj
+			if ok, proj := ConvertAggToProj(newAgg, newAgg.schema); ok {
```

Comment:
If it cannot be converted, we can break the loop here?

Reply:
yes, `ConvertAggToProj` will always be false in the loop